### PR TITLE
binutils: update to 2.42

### PR DESCRIPTION
**Pull request by jiegec**
Topic Description
-----------------

- bpftrace: bump REL
    Rebase patches
- binutils: update to 2.42

Package(s) Affected
-------------------

- binutils: 2.42
- bpftrace: 0.19.1-1

Security Update?
----------------

No

Build Order
-----------

```
#buildit binutils bpftrace
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`

**Secondary Architectures**

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

**Experimental Architectures**

- [x] MIPS R6 64-bit (Little Endian) `mips64r6el`

### CODE REVIEW

**Comment by jiegec** [comment]
bpftrace on loongarch64:

```shell
# bpftrace -e 'tracepoint:raw_syscalls:sys_enter { @[comm] = count(); }'
Attaching 1 probe...
^C

@[(sd-close)]: 2
@[httpd]: 6
@[packagekitd]: 7
@[openvpn]: 7
@[do-rsync]: 10
@[systemd-homed]: 15
@[systemd-resolve]: 15
@[dbus-daemon]: 20
@[rsync]: 26
@[systemd-machine]: 28
@[containerd]: 42
@[systemd-logind]: 43
@[bpftrace]: 46
@[systemd-journal]: 157
@[systemd]: 272
@[clash]: 582
@[sshd]: 869
@[popub-local]: 1514

```

**Comment by jiegec** [comment]
TODO: reconsider mips64 patch for loongson3/mips64r6el.

Update: done.

File: app-devel/binutils/spec [hunk 1]
```diff
@@ -1,4 +1,4 @@
-VER=2.41
+VER=2.42
```
Comment (MingcongBai) [inline app-devel/binutils/spec:1]: Please mark PKGBREAK if your intention is to rebuild the BPF-related packages for `libbfd`.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: app-devel/binutils/spec (hunk 1 above)
Comment (jiegec) [reply app-devel/binutils/spec:1]: Actually bpftrace only depends on libsframe from binutils, not libbfd where soname has been changed.

Comment (MingcongBai) [reply app-devel/binutils/spec:1]: > Actually bpftrace only depends on libsframe from binutils, not libbfd where soname has been changed.

So why rebuild?

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**